apps/omg_api/lib/state/core.ex

```diff
@@ -428,6 +435,67 @@ defmodule OMG.API.State.Core do
     {:ok, {event_triggers, db_updates, validities}, new_state}
   end
 
+  defp sigs_chope(<<>>), do: []
+  defp sigs_chope(<<sig::bytes-size(65), rest::binary>>), do: [sig | sigs_chope(rest)]
+
+  @spec in_flight_exits(in_flight_txs :: [in_flight_exit()], state :: t()) ::
+          {:ok, {[exit_event], [db_update]}, new_state :: t()}
+  def in_flight_exits(in_flight_txs, state) do
+    {db_updates_list_and_events, new_state} =
+      in_flight_txs
+      |> Enum.map_reduce(state, &in_flight_exit/2)
+
+    {event_triggers, db_updates} =
+      db_updates_list_and_events
+      |> Enum.unzip()
+      |> (fn {db_updates_list, event_triggers} -> {db_updates_list |> List.flatten(), event_triggers} end).()
+
+    {:ok, {event_triggers, db_updates}, new_state}
+  end
+
+  defp to_delete_db_update(utxos) do
+    utxos
+    |> Enum.map(fn Utxo.position(blknum, txindex, oindex) ->
+      {:delete, :utxo, {blknum, txindex, oindex}}
+    end)
+  end
+
+  defp in_flight_exit([tx_bytes, _, _, sigs], %Core{utxos: utxos} = state) do
+    {:ok, tx} = Transaction.decode(tx_bytes)
+    {:ok, tx_recover} = Transaction.Recovered.recover_from(%Transaction.Signed{raw_tx: tx, sigs: sigs_chope(sigs)})
+
+    {inputs, _invalid} =
+      tx_recover.signed_tx.raw_tx |> Transaction.get_inputs() |> Enum.split_with(&utxo_exists?(&1, state))
+
+    db_updates = to_delete_db_update(inputs)
+    new_state = %{state | utxos: Map.drop(utxos, inputs)}
+    event_trigger = %{in_flight_exits: %{utxo_pos: inputs}}
+
+    {{db_updates, event_trigger}, new_state}
```

Comment:
avoid the switch of the order of args/returns list, eg. put `event_triggers` consistently first